flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/sink/KafkaRecordSerializationSchema.java

```diff
@@ -48,6 +48,12 @@ default void open(
             SerializationSchema.InitializationContext context, KafkaSinkContext sinkContext)
             throws Exception {}
 
+    /**
+     * Closes this kafka serialization schema and releases any system resources associated with it.
+     * If the stream is already closed then invoking this method has no effect.
+     */
+    default void close() throws Exception {}
```

Comment:
Shouldn't this be invoked by the source reader?

Reply:
> Shouldn't this be invoked by the source reader?

You are right. I'll invoke close method in the `KafkaWriter`'s close method.